extension/intents/search/queryScript.js

```diff
@@ -92,23 +92,52 @@ this.queryScript = (function() {
   }
 
   communicate.register("searchResultInfo", message => {
-    const cards = findCards();
-    const searchHeaders = document.querySelectorAll("a > h3");
+    const GOOGLE_SELECTOR = "a > h3";
+    const DDG_SELECTOR = "#links .results_links_deep .result__a";
+    const BING_SELECTOR = ".b_algo h2 a";
+    let searchEngine = "google";
+    let cards;
+
+    if (message && message.searchEngine && message.searchEngine.toLowerCase()) {
+      searchEngine = message.searchEngine.toLowerCase();
```

Comment:
I think it would be easier to simply use `location.origin` to test here. Google has other names, like `google.ca` etc, but testing for `/google/i.test(location.origin)` or `/duckduckgo/i.test(location.origin)` should be reliable enough.

Reply:
Resolved using `location.origin` as you suggested.